src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,77 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = getImplementations(interfaceClazz);
+        if (clazz.isAnnotationPresent(Component.class)) {
+            Field[] fields = clazz.getDeclaredFields();
+            for (Field field : fields) {
+                if (field.isAnnotationPresent(Inject.class)) {
+                    Object fieldInstance = getInstance(field.getType());
+                    clazzImplementationInstance = createNewInstance(clazz);
+                    field.setAccessible(true);
+                    try {
+                        field.set(clazzImplementationInstance, fieldInstance);
+                    } catch (IllegalAccessException e) {
+                        throw new RuntimeException("Cant initialize field value. "
+                                + "Class: " + clazz.getName()
+                                + "Field: " + field.getName(), e);
+                    }
+                }
+            }
+            if (clazzImplementationInstance == null) {
+                clazzImplementationInstance = createNewInstance(clazz);
+            }
+            return clazzImplementationInstance;
+        }
+        throw new RuntimeException("Unsupported class. Class: " + clazz.getName());
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (InvocationTargetException
+                | InstantiationException
+                | IllegalAccessException
+                | NoSuchMethodException e) {
```

Comment:
```suggestion
        } catch (ReflectiveOperationException e) {
```
we can use this for simplicity